test/e2e/conftest.py

```
@@ -85,7 +86,21 @@ def container_registry():
             check=True,
         )
         # fmt: on
-        time.sleep(2)
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+        start_time = time.time()
+        while time.time() - start_time < 30:
+            try:
+                resp = requests.get(
+                    f"https://localhost:{registry_port}/v2/",
+                    verify=False,
```

Ruff (0.15.15)

[error] Probable use of `requests` call with `verify=False` disabling SSL certificate checks

(S501)